### PR TITLE
docs: exposing single server grafana with -p

### PR DESCRIPTION
**Pull request by uwedeportivo**
Remove socat + sshuttle contraption and use simple -p 127.0.0.1: syntax for docker run

### CODE REVIEW

File: doc/admin/monitoring_and_tracing.md [hunk 1]
```diff
@@ -33,28 +33,10 @@ kubectl port-forward svc/grafana 3370:30070
...
 
 ```shell script
-docker exec -t XXX_CONTAINER_ID_XXXX apk add --no-cache socat
-socat tcp-listen:3370,reuseaddr,fork system:"docker exec -i 7298cae012eb socat stdio 'tcp:localhost:3370'"
-```
-
-On your local machine start a `sshuttle` session to your remote server
-
-```shell script
-sshuttle -r xxx_user_xxx@sxxx_remote_server_xxx 0/0
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]: They still need `sshuttle` for accessing this on remote machines

File: doc/admin/monitoring_and_tracing.md [hunk 2]
```diff
@@ -33,28 +33,10 @@ kubectl port-forward svc/grafana 3370:30070
...
-```
-
-### Single-container server deployments (with firewall)
-
-If you are running Sourcegraph as single server behind a firewall you can add an additional publish flag for the Grafana port 3370:
-
-```shell script
-docker run --publish 7080:7080 --publish 2633:2633 --publish 3370:3370 \
+docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]: You can do this in a different PR, but we could do this everywhere we suggest a `docker run` command like on https://docs.sourcegraph.com/ -- any reason you don't want to do that?

File: doc/admin/monitoring_and_tracing.md (hunk 1 above)
Comment (uwedeportivo) [reply doc/admin/monitoring_and_tracing.md]: oh yeah

File: doc/admin/monitoring_and_tracing.md (hunk 2 above)
Comment (slimsag) [reply doc/admin/monitoring_and_tracing.md]: (then this section would just describe what to visit in your browser, and how to use `sshuttle` if running Sourcegraph on a remote server)

Comment (uwedeportivo) [reply doc/admin/monitoring_and_tracing.md]: sure, i'll scrub and add it to all

File: doc/admin/monitoring_and_tracing.md (hunk 1 above)
Comment (uwedeportivo) [reply doc/admin/monitoring_and_tracing.md]: done

File: doc/admin/monitoring_and_tracing.md (hunk 2 above)
Comment (uwedeportivo) [reply doc/admin/monitoring_and_tracing.md]: done

File: doc/admin/monitoring_and_tracing.md [hunk 3]
```diff
@@ -33,32 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
 ```shell script
-docker exec -t XXX_CONTAINER_ID_XXXX apk add --no-cache socat
-socat tcp-listen:3370,reuseaddr,fork system:"docker exec -i 7298cae012eb socat stdio 'tcp:localhost:3370'"
+docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \
+  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
+  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
 ```
 
-On your local machine start a `sshuttle` session to your remote server
+then http://localhost:3370/-/debug/grafana will just work. If you are running it on a remote server you can use a tool
```
Comment (ryan-blunden) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.

Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana . If Sourcegraph is deployed to a remote server, a tool 
```

File: doc/admin/monitoring_and_tracing.md [hunk 4]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
+docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \
+  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
+  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
 ```
 
-### Single-container server deployments (with firewall)
+For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.
 
-If you are running Sourcegraph as single server behind a firewall you can add an additional publish flag for the Grafana port 3370:
+Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, a tool
```
Comment (ryan-blunden) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, then access via an SSH tunnel using a tool
```

File: doc/admin/monitoring_and_tracing.md [hunk 5]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
+  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
+  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
 ```
 
-### Single-container server deployments (with firewall)
+For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.
 
-If you are running Sourcegraph as single server behind a firewall you can add an additional publish flag for the Grafana port 3370:
+Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, a tool 
+like [sshuttle](https://github.com/sshuttle/sshuttle) to establish a secure connection to that URL.
```
Comment (ryan-blunden) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
such as [sshuttle](https://github.com/sshuttle/sshuttle) is required to establish a secure connection to Grafana.
```

File: doc/admin/monitoring_and_tracing.md [hunk 6]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
+  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
 ```
 
-### Single-container server deployments (with firewall)
+For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.
 
-If you are running Sourcegraph as single server behind a firewall you can add an additional publish flag for the Grafana port 3370:
+Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, a tool 
+like [sshuttle](https://github.com/sshuttle/sshuttle) to establish a secure connection to that URL. 
+On your local machine start a `sshuttle` session to your remote server
```
Comment (ryan-blunden) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
To access the remote server using `sshuttle` from your local machine
```

File: doc/admin/monitoring_and_tracing.md [hunk 7]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
-If you are running Sourcegraph as single server behind a firewall you can add an additional publish flag for the Grafana port 3370:
+Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, a tool 
+like [sshuttle](https://github.com/sshuttle/sshuttle) to establish a secure connection to that URL. 
+On your local machine start a `sshuttle` session to your remote server	
 
-```shell script
-docker run --publish 7080:7080 --publish 2633:2633 --publish 3370:3370 \
-  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
-  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
+```shell script
```
Comment (ryan-blunden) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
```sh
```

File: doc/admin/monitoring_and_tracing.md [hunk 8]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
+Therefore, if accessing Grafana locally, the URL will be http://localhost:3370/-/debug/grafana. If Sourcegraph is deployed to a remote server, a tool 
+like [sshuttle](https://github.com/sshuttle/sshuttle) to establish a secure connection to that URL. 
+On your local machine start a `sshuttle` session to your remote server	
 
-```shell script
-docker run --publish 7080:7080 --publish 2633:2633 --publish 3370:3370 \
-  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
-  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
+```shell script	
+sshuttle -r xxx_user_xxx@sxxx_remote_server_xxx 0/0
```
Comment (ryan-blunden) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
sshuttle -r user@host 0/0
```

You can also use `ssh` to establish an SSH tunnel (although `sshuttle` is much more flexible and reliable):

```sh
SSH TUNNEL COMMAND GOES HERE
```

Comment (ryan-blunden) [reply doc/admin/monitoring_and_tracing.md]: It's always good if possible, to provide a command that does not require the user to install a new tool. Obviously, if that's not the case for any reason, this doesn't apply.

File: doc/admin/monitoring_and_tracing.md [hunk 9]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
-socat tcp-listen:3370,reuseaddr,fork system:"docker exec -i 7298cae012eb socat stdio 'tcp:localhost:3370'"
-```
-
-On your local machine start a `sshuttle` session to your remote server
-
-```shell script
-sshuttle -r xxx_user_xxx@sxxx_remote_server_xxx 0/0
+docker run --publish 7080:7080 --publish 2633:2633 --publish 127.0.0.1:3370:3370 \
+  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
+  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]: Why are we still providing the docker run command here, and not the sshuttle command?

File: doc/admin/monitoring_and_tracing.md [hunk 10]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
 
 ### Single-container server deployments
 
-If you are running Sourcegraph as a single server in a droplet or otherwise exposed to the open internet then publishing
-an additional docker port is not recommended since the embedded Grafana runs in anonymous mode. In this case we suggest using
-the tools `sshuttle` and `socat` to create a secure connection. Log into your remote server using ssh 
-and run these commands: 
+If you are running Sourcegraph as single server locally:
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
If you are running Sourcegraph as a single-container server deployment:
```

File: doc/admin/monitoring_and_tracing.md [hunk 11]
```diff
@@ -33,30 +33,22 @@ kubectl port-forward svc/grafana 3370:30070
...
+such as [sshuttle](https://github.com/sshuttle/sshuttle) is required to establish a secure connection to Grafana.
+To access the remote server using `sshuttle` from your local machine:
 
-```shell script
-docker run --publish 7080:7080 --publish 2633:2633 --publish 3370:3370 \
-  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
-  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
+```sh
+sshuttle -r user@host 0/0
 ```
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
Then simply visit ... in your browser.
```

File: doc/admin/monitoring_and_tracing.md (hunk 9 above)
Comment (uwedeportivo) [reply doc/admin/monitoring_and_tracing.md]: seemed like a good place for a reminder how to run it :-). i'll remove it in a bit. am in the middle of some graphql metrics stuff on another branch

File: doc/admin/monitoring_and_tracing.md [hunk 12]
```diff
@@ -33,31 +33,16 @@ kubectl port-forward svc/grafana 3370:30070
 
 ### Single-container server deployments
 
-If you are running Sourcegraph as a single server in a droplet or otherwise exposed to the open internet then publishing
-an additional docker port is not recommended since the embedded Grafana runs in anonymous mode. In this case we suggest using
-the tools `sshuttle` and `socat` to create a secure connection. Log into your remote server using ssh 
-and run these commands: 
+For simplicity, Garafana does not require authentication, as the port binding of 3370 is limited to connections from localhost.
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
For simplicity, Garafana does not require authentication, as the port binding of 3370 is restricted to connections from localhost only.
```

File: doc/admin/monitoring_and_tracing.md [hunk 13]
```diff
@@ -33,31 +33,16 @@ kubectl port-forward svc/grafana 3370:30070
...
-If you are running Sourcegraph as single server behind a firewall you can add an additional publish flag for the Grafana port 3370:
-
-```shell script
-docker run --publish 7080:7080 --publish 2633:2633 --publish 3370:3370 \
-  --rm --volume ~/.sourcegraph/config:/etc/sourcegraph \
-  --volume ~/.sourcegraph/data:/var/opt/sourcegraph sourcegraph/server:3.9.3
+```sh
+sshuttle -r user@host 0/0
 ```
+Then simply visit ... in your browser.
```
Comment (slimsag) [inline doc/admin/monitoring_and_tracing.md]:
```suggestion
Then simply visit http://host:3370 in your browser.
```

I meant for you to replace ... with whatever is relevant, is this the correct URL or would it need /-/debug/ etc?

Comment (slimsag) [reply doc/admin/monitoring_and_tracing.md]: I see it redirects automatically, so the URL I suggested here is fine.
